### PR TITLE
examples/camera: Fix some issues and update license

### PR DESCRIPTION
**Pull request by SPRESENSE**
## Summary
Fix some issues and update license to Apache2.
- Fix  failure to open the video driver.
- Change format from YUV422 to RGB565,
  and add captured image output to LCD.
- An application cannot exit by infinite loop.
- In video-only mode without capturing still images,
  do not allocate capture memory for still images to
  reduce the memory usage.

## Impact
examples/camera

## Testing
Tested by spresense:example_camera

### CODE REVIEW

**Comment by jerpelea** [comment]
@SPRESENSE 
apps/examples/camera/camera_main.c:639:44: error: Operator/assignment must be followed with whitespace

**Comment by jerpelea** [comment]
@SPRESENSE 
Error: camera_main.c:488:9: error: too few arguments to function 'video_initialize'

File: examples/camera/camera_main.c [hunk 1]
```diff
@@ -525,74 +536,182 @@ int main(int argc, FAR char *argv[])
...
+            if (ret != OK)
+              {
+                goto exit_this_app;
+              }
+
+            if (!is_eternal)
+              {
+                gettimeofday(&now, NULL);
+                timersub(&now, &start, &delta);
+                if (timercmp(&delta, &wait, >))
```
Comment (SPRESENSE) [inline examples/camera/camera_main.c:639]: @jerpelea 
The general implementation of `timercmp()` causes an nxstyle error. How can I fix this?

Comment (jerpelea) [reply examples/camera/camera_main.c:639]: -                if (timercmp(&delta, &wait, >))
+                if (timercmp(&delta,
+                             &wait,
+                             >
+                             ))

Comment (jerpelea) [reply examples/camera/camera_main.c:639]:
```suggestion
-                if (timercmp(&delta, &wait, >))
+                if (timercmp(&delta,
+                             &wait,
+                             >
+                             ))

```

Comment (SPRESENSE) [reply examples/camera/camera_main.c:639]: @jerpelea thank you for your suggestion.
The readability is lower, but should I change to fix it as you suggested?

**Comment by jerpelea** [comment]
I have no objection if we merge it as it is

File: examples/camera/camera_main.c (hunk 1 above)
Comment (pkarashchenko) [reply examples/camera/camera_main.c:639]: Let's keep the current version and ignore style error.

**Pull request merged**